### PR TITLE
feat(backend): optional pagination for user query

### LINKED ISSUE #1443: 🚀 [Feature] Backend: Users Query Pagination
## 🚀 Feature

So far all the users are returned bu the users query. I only want to receive a certain number of users and have the possibility to fetch more users if needed.

## To Dos

Create a pagination input:
```graphql
input Pagination {
  take: Int!
  skip: Int!
}
```
with defaults `take = 5` and `skip = 0`.

```graphql
query users($pagination: Pagination) {
  users(pagination: $Pagination) {
    id
    name
    username    
  }
}
```

Unit tests that show that with a lot of users in the database the returned user list is working correctly with different pagination values.

Hint: https://www.prisma.io/docs/orm/prisma-client/queries/pagination

### PR DESCRIPTION
**Pull request by trinity2701**
### Issues
based on #1518 

### Todo

- [X] None

### CODE REVIEW

File: backend/src/graphql/resolvers/UserResolver.spec.ts [hunk 1]
```diff
@@ -100,6 +100,227 @@ describe('UserResolver', () => {
           })
         })
       })
+
+      describe('include search string', () => {
```
Comment (roschaefer) [inline backend/src/graphql/resolvers/UserResolver.spec.ts:104]:
```suggestion
      describe('with a given `$searchString` resolver argument', () => {
```
image

File: backend/src/graphql/resolvers/UserResolver.spec.ts [hunk 2]
```diff
@@ -100,6 +100,227 @@ describe('UserResolver', () => {
...
+              },
+              {
+                name: 'Tomás Schmit',
+                username: 'Schmit',
+              },
+            ],
+          })
+        })
+
+        it('returns the correct users for "tom"', async () => {
```
Comment (roschaefer) [inline backend/src/graphql/resolvers/UserResolver.spec.ts:128]:
```suggestion
        it('filters users by `username` and `name`', async () => {
```

File: backend/src/graphql/resolvers/UserResolver.spec.ts [hunk 3]
```diff
@@ -100,6 +100,227 @@ describe('UserResolver', () => {
...
+                    },
+                  ],
+                },
+                errors: undefined,
+              },
+            },
+          })
+        })
+
+        it('returns the correct users for "TOM"', async () => {
```
Comment (roschaefer) [inline backend/src/graphql/resolvers/UserResolver.spec.ts:180]:
```suggestion
        it('ignores upper case and lower case', async () => {
```

File: backend/src/graphql/resolvers/UserResolver.spec.ts [hunk 4]
```diff
@@ -100,6 +100,227 @@ describe('UserResolver', () => {
...
+                    },
+                  ],
+                },
+                errors: undefined,
+              },
+            },
+          })
+        })
+
+        it('returns the correct users for "sch"', async () => {
```
Comment (roschaefer) [inline backend/src/graphql/resolvers/UserResolver.spec.ts:232]: I don't see the point of this test case - what are you testing? Substring was considered above.

File: backend/src/graphql/resolvers/UserResolver.spec.ts [hunk 5]
```diff
@@ -100,6 +100,227 @@ describe('UserResolver', () => {
...
+                    },
+                  ],
+                },
+                errors: undefined,
+              },
+            },
+          })
+        })
+
+        it('returns the correct users for "tomas"', async () => {
```
Comment (roschaefer) [inline backend/src/graphql/resolvers/UserResolver.spec.ts:284]:
```suggestion
        it('ignores special characters e.g. `á` is like `a`', async () => {
```

**Review by roschaefer — APPROVED** [review]
Looks good, but has nothing to do with pagination. I will remove the linked issue.

search

File: backend/src/graphql/resolvers/UserResolver.spec.ts (hunk 4 above)
Comment (Mogge) [reply backend/src/graphql/resolvers/UserResolver.spec.ts:232]: I do not agree. We tested for strings starting with the search string. This test proves, that the search-string also matches in the middle of the string

Comment (roschaefer) [reply backend/src/graphql/resolvers/UserResolver.spec.ts:232]: OK! "tom" is indeed the prefix. Then:
```suggestion
        it('filters by substring' async () => {
```

**Comment by roschaefer** [comment]
> @roschaefer The code you reviewed is already in master.

How come? This PR here is in "Draft" mode, I looks not being merged. If the changes are already in `master` we should see the PR getting auto-closed when I hit "rebase"...

**Comment by roschaefer** [comment]
Ahh, merge conflicts. So, can we close this PR then?

**Pull request closed without merging**